### ISSUE TITLE
Helidon and JPA weave fails with docker

### ISSUE THREAD

**Issue by rafabene** [oracle/helidon#2128]
## Environment Details
* Helidon Version: 2.0.1
* Helidon MP
* JDK version: 11
* OS: MacOS
* Docker version (if applicable): 19.03.8

----------
I've enabled `eclipselink-maven-plugin` as instructed at https://helidon.io/docs/v2/#/mp/guides/09_jpa Then my docker build fails with the following error message:

> [INFO] --- eclipselink-maven-plugin:2.7.1.1:weave (weave) @ payment ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  05:09 min
[INFO] Finished at: 2020-07-02T15:57:45Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.ethlo.persistence.tools:eclipselink-maven-plugin:2.7.1.1:weave (weave) on project payment: Source directory /helidon/target/classes does not exist -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
ERROR: Service 'payment' failed to build: The command '/bin/sh -c mvn package -DskipTests' returned a non-zero code: 1

## Steps to reproduce
- Create a Helidon MP project using 2.0.1
- Enable JPA using the instructions at https://helidon.io/docs/v2/#/mp/guides/09_jpa
- Perform a docker build using the default Dockerfile

**Comment by rafabene** [comment]
Someone else faced the same problem: https://www.albinhasani.net/helidon-mp

**Comment by ljnelson** [comment]
The root issue appears to be our usage of `-DskipTests` in the first stage of our Dockerfile builds.  It should be `-Dmaven.test.skip` to also skip test compilation, which is also not necessary for that stage.

**Comment by romain-grecourt** [comment]
You can also use `-Declipselink.weave.skip=true`

**Comment by barchetta** [comment]
Note that the above PR merge does not fix this issue, and `eclipselink.weave.skip` is not supported by the version of the eclipse plugin we are using (2.7.1.1). So we need to upgrade to 2.7.4 or newer. See PR #2138 .

**Comment by barchetta** [comment]
Work around for 2.0.1 and earlier:

1. Modify your project's `pom.xml`:

    Force newer version of eclipselink:
    ```
    <properties>
        <version.plugin.eclipselink>2.7.5.1</version.plugin.eclipselink>
        <version.lib.eclipselink>2.7.5</version.lib.eclipselink>
    </properties>
    ```

    Remove `<version>2.7.1.1</version>` from `eclipselink-maven-plugin` plugin section if it is there.

2. Update Dockerfile to skip weaving during dependency priming build. 

    Change *first* invocation of `mvn` from

    ```
    RUN mvn package -DskipTests
    ```
    to

    ```
    RUN mvn package -Dmaven.test.skip -Declipselink.weave.skip
    ```

**Comment by barchetta** [comment]
Fixed by #2146
